### PR TITLE
chore(buffers): remove LevelDB-based disk_v1 buffer impl

### PR DESCRIPTION
**Pull request by tobz**
This PR entirely removes the old LevelDB-based `disk_v1` disk buffer implementation.

One unrelated tweak is removing the `parking_lot` dependency and using `std::sync::Mutex`, as when checking for unused dependencies related to `disk_v1`, `cargo-udeps` pointed out that `parking_lot` was unused... since it was actually only used in the tests.

### CODE REVIEW

File: lib/vector-core/src/event/test/serialization.rs [hunk 1]
```diff
@@ -17,100 +15,6 @@ fn decode_value<T: Encodable, B: Buf + Clone>(buffer: B) -> T {
     T::decode(T::get_metadata(), buffer).expect("decoding should not fail")
 }
 
-#[test]
-fn encodable_must_decode_single_eventarray_while_leveldb_disk_buffer_still_present() {
-    // REVIEWERS: Be aware, if this is being removed or changed, the only acceptable context is
-    // LevelDB-based disk buffers being removed, or some other extenuating circumstance that must be
-    // explained.
```
Comment (spencergilbert) [inline lib/vector-core/src/event/test/serialization.rs:24]: 🎉

**Review by spencergilbert — APPROVED** [review]
Huzzah!

**Review by bruceg — APPROVED** [review]
🎉

**Comment by jszwedko** [comment]
Just noticing that this PR seems to be missing an upgrade note. I'll add one as part of preparing the 0.28.0 release.

**Pull request merged**